File: semester 4/MPP/JFTS/FestivalService/src/main/java/ro/mpp/service/IFestivalService.java
package ro.mpp.service;

import ro.mpp.domain.Artist;
import ro.mpp.domain.Show;
import ro.mpp.domain.ShowArtist;
import ro.mpp.domain.Ticket;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

public interface IFestivalService {
    // ---------- SHOW-ARTIST
    public List<ShowArtist> findByDate(LocalDate date);
    public List<ShowArtist> findAll();
    public Optional<Show> findById(int id);

    // ---------- TICKET
    public Optional<Ticket> sellTicket(Show show, String buyerName, int seats);
    public boolean modifyTicket(int ticketId, int seats);
}

File: semester 4/MPP/C#FTS/C#FTS/Service/IFestivalService.cs
using C_FTS.Domain;
using System;
using System.Collections.Generic;
using System.Text;

namespace C_FTS.Service
{
    public class TicketModifier : ApplicationException
    {
        public TicketModifier() { }
        public TicketModifier(string message) : base(message) { }
        public TicketModifier(string message, Exception innerException) : base(message, innerException) { }
    }

    public interface IFestivalService
    {
        // ---------- SHOW-ARTIST
        public List<ShowArtist> FindByDate(DateTime date);
        public List<ShowArtist> FindAll();

        // ---------- TICKET
        public Ticket? SellTicket(Show show, string buyerName, int seats);
        public bool ModifyTicket(int ticketId, int seats);
    }
}
